src/core/api/__tests__/get_player_state.test.ts

```diff
@@ -0,0 +1,227 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from "chai";
+import config from "../../../config";
+import getLoadedContentState from "../get_player_state";
+
+describe("API - getLoadedContentState", () => {
+  it("should always return ENDED if mediaElement.ended is true", () => {
+    const fakeProps = {
+      ended: true,
+      duration: 100000,
+      currentTime: 0,
+    };
+    const mediaElement = fakeProps as HTMLMediaElement;
+
+    // we can just do every possibility here
+    expect(getLoadedContentState(mediaElement, true, null))
+      .to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, false, null))
+      .to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, true, {
+      reason: "seeking",
+    })).to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, true, {
+      reason: "not-ready",
+    })).to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, true, {
+      reason: "buffering",
+    })).to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, false, {
+      reason: "seeking",
+    })).to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, false, {
+      reason: "not-ready",
+    })).to.equal("ENDED");
+    expect(getLoadedContentState(mediaElement, false, {
+      reason: "buffering",
+    })).to.equal("ENDED");
+  });
+
+  it("should be PLAYING if not stalled nor ended and if not paused", () => {
+    const fakeProps = {
+      ended: false,
+      duration: 10,
+      currentTime: 10, // worst case -> currentTime === duration
+    };
+    const mediaElement = fakeProps as HTMLMediaElement;
+    expect(getLoadedContentState(mediaElement, true, null))
+      .to.equal("PLAYING");
+  });
+
+  it("should be PAUSED if not stalled nor ended and if paused", () => {
+    const fakeProps = {
+      ended: false,
+      duration: 10,
+      currentTime: 10, // worst case -> currentTime === duration
+    };
+    const mediaElement = fakeProps as HTMLMediaElement;
+    expect(getLoadedContentState(mediaElement, false, null))
+      .to.equal("PAUSED");
+  });
+
+  it("should be BUFFERING if not ended and stalled because of bufferring", () => {
+    const fakeProps = {
+      ended: false,
+      duration: 10,
+      currentTime: 5,
+    };
+    const mediaElement = fakeProps as HTMLMediaElement;
+    expect(getLoadedContentState(mediaElement, true, {
+      reason: "buffering",
+    })).to.equal("BUFFERING");
+    expect(getLoadedContentState(mediaElement, false, {
+      reason: "buffering",
+    })).to.equal("BUFFERING");
+  });
+
+  it("should be BUFFERING if not ended and stalled because of `not-ready`", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 77. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=414&issues=AWgOMLyQ_MPaBDn0sxTa&open=AWgOMLyQ_MPaBDn0sxTa)

[ISSUE_KEY]: AWgOMLyQ_MPaBDn0sxTa